src/gst.rs

```diff
@@ -84,21 +98,47 @@ impl RtspServer {
         Ok(maybe_app_src)
     }
 
-    pub fn set_credentials(&mut self, user_pass: Option<(&str, &str)>) -> Result<()> {
-        let auth = user_pass.map(|(user, pass)| {
-            let auth = RTSPAuth::new();
-            /*
-            let perm = RTSPToken::new(
-                ...
-            );
-            auth.add_basic(RTSPAuth::make_basic(user, pass).as_str(), &perm);
-            */
-            // TODO TLS https://thiblahute.github.io/GStreamer-doc/gst-rtsp-server-1.0/rtsp-server.html?gi-language=c
-            auth
-        });
+    pub fn set_credentials(&self, user :&str, pass :&str) -> Result<()> {
+        let auth = match self.server.get_auth() {
+            Some(x) => x,
+            None =>  RTSPAuth::new(),
+        };
+
+        if ! user.is_empty() && ! pass.is_empty() {
+            info!("Setting credentials for user {}", user);
+            debug!("Password is {}", pass);
```

Comment:
Never print passwords into a log. Change "setting credentials" to `debug` log level.

Reply:
Sure sounds like a good plan